pkg/apis/servicecatalog/types.go

```diff
@@ -266,10 +266,16 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// Parameters is set of properties to be
 	// passed to the underlying broker.
 	Parameters *runtime.RawExtension
 
+	// Optional: AlphaSecretParameters is reference to the secret object containing
+	// sensitive information to pass to the underlying broker. This may be
+	// empty if no secret object is specified.
+	// +optional
+	AlphaSecretParameters *v1.LocalObjectReference
```

Comment:
If so, do we want to allow this?  While conceptually being able to pull data from other places sounds nice, I wonder if that's really wise to do since that means the data we're pulling from probably isn't created especially for the svc-cat processing.  Which means it might be changed over time, or worse, transient and deleted.  What does it mean if its changed? Should that force a new instance/binding request?  Do we even want to allow people to change this data at all once the instance or binding has been created w/o explicitly telling us the new data - meaning require an explicit PATCH to the Instance/Binding resource so we can then trigger the correct OSBAPI call (or block the PATCH if necessary)?

What I'm pondering is whether, at least for now, we should force people to explicitly provide us the data (either directly or a ref to a single secret for secure params) - then we know its likely just for us and won't be changed.  I'm also wondering if it wouldn't be better to go one step further and require all of the data to be included in the creation of the Instance/Binding and then we create the secret (for the private bits) automatically for them - then we control the lifecycle of that secret, can do conflict checking, its clear its not for other uses and the user is then required to explicitly tell us when the data has changed via a PATCH.  We can null-out the secret data from the InstanceSpec so its not visible after the secret has been created.

I guess we could do this copy-on-create logic even with what Paul is proposing too...

Thoughts?